contrib/udfs/src/main/java/org/apache/drill/exec/udfs/DistributionFunctions.java

```diff
@@ -363,4 +365,44 @@ public void eval() {
       }
     }
   }
+  @SuppressWarnings("unused")
+  @FunctionTemplate(name = "median",
```

Comment:
Have you considered using the same Freemarker template approach followed by the other aggregate statistical functions? If you can rebase on #2804 then I'm hopeful that any preferred cast ambiguity for a VARCHAR argument will be resolved.

Reply:
@jnturton I went ahead and did this with Freemarker templates as I think the approach is generally better anyway.